kv/kvserver: TestRangefeedCheckpointsRecoverFromLeaseExpiration failed

kv/kvserver.TestRangefeedCheckpointsRecoverFromLeaseExpiration [failed](https://tanzanite.cluster.engflow.com/invocations/default/6d51a067-400a-4719-b2a2-2ad4da949e87?testReportRun=2&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [ac3c5f2e66f0bb3c992c719389f15d855248a781](https://github.com/cockroachdb/cockroach/commits/ac3c5f2e66f0bb3c992c719389f15d855248a781):


```
=== RUN   TestRangefeedCheckpointsRecoverFromLeaseExpiration
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3397693360
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_rangefeed_test.go:1635: 
        	Error Trace:	pkg/kv/kvserver/replica_rangefeed_test.go:1635
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRangefeedCheckpointsRecoverFromLeaseExpiration
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3397693360
--- FAIL: TestRangefeedCheckpointsRecoverFromLeaseExpiration (16.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedCheckpointsRecoverFromLeaseExpiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestRangefeedCheckpointsRecoverFromLeaseExpiration [failed](https://tanzanite.cluster.engflow.com/invocations/default/7902003c-cafa-473c-a798-fa4bdb67f21f?testReportRun=3&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a0faa4e779ac9e54d17e8400141b0aa472887974](https://github.com/cockroachdb/cockroach/commits/a0faa4e779ac9e54d17e8400141b0aa472887974):


```
=== RUN   TestRangefeedCheckpointsRecoverFromLeaseExpiration
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3316682907
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_rangefeed_test.go:1635: 
        	Error Trace:	pkg/kv/kvserver/replica_rangefeed_test.go:1635
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRangefeedCheckpointsRecoverFromLeaseExpiration
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3316682907
--- FAIL: TestRangefeedCheckpointsRecoverFromLeaseExpiration (16.25s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedCheckpointsRecoverFromLeaseExpiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestRangefeedCheckpointsRecoverFromLeaseExpiration [failed](https://tanzanite.cluster.engflow.com/invocations/default/874184ef-0e07-4504-b611-8aa24e9332d4?testReportRun=3&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [47699f3887ad5d1b8c7c5905eb5c49628aa59bbe](https://github.com/cockroachdb/cockroach/commits/47699f3887ad5d1b8c7c5905eb5c49628aa59bbe):


```
=== RUN   TestRangefeedCheckpointsRecoverFromLeaseExpiration
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration2688618490
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_rangefeed_test.go:1635: 
        	Error Trace:	pkg/kv/kvserver/replica_rangefeed_test.go:1635
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRangefeedCheckpointsRecoverFromLeaseExpiration
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration2688618490
--- FAIL: TestRangefeedCheckpointsRecoverFromLeaseExpiration (16.05s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedCheckpointsRecoverFromLeaseExpiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestRangefeedCheckpointsRecoverFromLeaseExpiration [failed](https://tanzanite.cluster.engflow.com/invocations/default/4b257292-239c-4263-8c6e-9a89b6442bd2?testReportRun=1&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [6e405b0552362bad6f0df5261376193fbf8b98cd](https://github.com/cockroachdb/cockroach/commits/6e405b0552362bad6f0df5261376193fbf8b98cd):


```
=== RUN   TestRangefeedCheckpointsRecoverFromLeaseExpiration
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3551002081
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_rangefeed_test.go:1635: 
        	Error Trace:	pkg/kv/kvserver/replica_rangefeed_test.go:1635
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 0
        	Test:       	TestRangefeedCheckpointsRecoverFromLeaseExpiration
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestRangefeedCheckpointsRecoverFromLeaseExpiration3551002081
--- FAIL: TestRangefeedCheckpointsRecoverFromLeaseExpiration (16.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedCheckpointsRecoverFromLeaseExpiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/replica_rangefeed_test.go
+++ b/pkg/kv/kvserver/replica_rangefeed_test.go
@@ -1631,8 +1631,12 @@ func TestRangefeedCheckpointsRecoverFromLeaseExpiration(t *testing.T) {
 	log.Infof(ctx, "test waiting for another checkpoint")
 	ts2 := n1.Clock().Now()
 	waitForCheckpoint(ts2)
-	nudged := atomic.LoadInt64(&nudgeSeen)
-	require.Equal(t, int64(1), nudged)
+	testutils.SucceedsSoon(t, func() error {
+		if atomic.LoadInt64(&nudgeSeen) != int64(1) {
+			return errors.Errorf("nudge not seen yet")
+		}
+		return nil
+	})
 
 	// Check that n2 renewed its lease, like the test intended.
 	// Unfortunately this is flaky and it's not so clear how to fix it.

